### PR TITLE
PIM-4004: added a detacher in product writer

### PR DESCRIPTION
**Pull request by solivier**

### CODE REVIEW

**Comment by jjanvier** [comment]
change your BC break so

if no memory leak, GTM :p

File: spec/Pim/Bundle/BaseConnectorBundle/Writer/Doctrine/ProductWriterSpec.php [hunk 1]
```diff
@@ -14,9 +15,10 @@ class ProductWriterSpec extends ObjectBehavior
     function let(
         MediaManager $mediaManager,
         VersionManager $versionManager,
-        ProductSaver $productSaver
+        ProductSaver $productSaver,
```
Comment (jjanvier) [inline spec/Pim/Bundle/BaseConnectorBundle/Writer/Doctrine/ProductWriterSpec.php]: why don't you use a SaverInterface ?

File: src/Pim/Bundle/BaseConnectorBundle/Writer/Doctrine/ProductWriter.php [hunk 2]
```diff
@@ -101,6 +108,8 @@ public function write(array $items)
         }
         $this->mediaManager->handleAllProductsMedias($items);
         $this->productSaver->saveAll($items, ['recalculate' => false]);
+
+        $this->detacher->detachAll($items);
```
Comment (nidup) [inline src/Pim/Bundle/BaseConnectorBundle/Writer/Doctrine/ProductWriter.php]: Drop useless blank line update the changelog then ok :)

File: spec/Pim/Bundle/BaseConnectorBundle/Writer/Doctrine/ProductWriterSpec.php (hunk 1 above)
Comment (nidup) [inline spec/Pim/Bundle/BaseConnectorBundle/Writer/Doctrine/ProductWriterSpec.php]: Agree, SaverInterface FTW

**Pull request merged**